rmw_connext_dynamic_cpp/src/functions.cpp

```diff
@@ -1900,25 +1900,25 @@ rmw_destroy_subscription(rmw_node_t * node, rmw_subscription_t * subscription)
   if (custom_subscription_info) {
     DDSDynamicDataTypeSupport * ddts = custom_subscription_info->dynamic_data_type_support_;
     if (ddts) {
-      std::string type_name = _create_type_name(custom_subscription_info->members_, "msg");
-      // Cannot unregister type here, in case another topic is using the same type.
-      // Should be cleaned up when the node is destroyed.
-      // If we figure out a way to unregister types when they are not being used, then we can
-      // add this code back in:
-      /*
-         if (ddts->unregister_type(participant, type_name.c_str()) != DDS_RETCODE_OK) {
-         RMW_SET_ERROR_MSG(("failed to unregister type: " + type_name).c_str());
-         return RMW_RET_ERROR;
-         }
-       */
       if (custom_subscription_info->dynamic_data) {
         if (ddts->delete_data(custom_subscription_info->dynamic_data) != DDS_RETCODE_OK) {
           RMW_SET_ERROR_MSG("failed to delete dynamic data");
           return RMW_RET_ERROR;
         }
+        custom_subscription_info->dynamic_data = nullptr;
+      }
+      std::string type_name = _create_type_name(custom_subscription_info->members_, "msg");
+      // TODO Cannot unregister and free type here, in case another topic is using the same type.
```

Comment:
I think it was William but I would rather not bother because now multiple people have updated the block already. Important is that we find this again - therefore I added the `TODO`.